File: src/main/java/com/matag/game/turn/action/leave/PutIntoGraveyardService.java
package com.matag.game.turn.action.leave;

import java.util.List;

import org.springframework.stereotype.Component;

import com.matag.game.cardinstance.CardInstance;
import com.matag.game.status.GameStatus;

@Component
public class PutIntoGraveyardService {
  public void putIntoGraveyard(GameStatus gameStatus, List<CardInstance> cards) {
    for (CardInstance card : cards) {
      var owner = gameStatus.getPlayerByName(card.getOwner());
      card.resetAllModifiers();
      owner.getGraveyard().addCard(card);
    }
  }

  public void putIntoGraveyard(GameStatus gameStatus, CardInstance cardInstance) {
    putIntoGraveyard(gameStatus, List.of(cardInstance));
  }
}

File: src/main/kotlin/com/matag/game/turn/action/leave/PutIntoGraveyardService.kt
package com.matag.game.turn.action.leave

import com.matag.game.cardinstance.CardInstance
import com.matag.game.status.GameStatus
import org.springframework.stereotype.Component

@Component
class PutIntoGraveyardService {
    fun putIntoGraveyard(gameStatus: GameStatus, cards: List<CardInstance>) {
        for (card in cards) {
            val owner = gameStatus.getPlayerByName(card.owner)
            card.resetAllModifiers()
            owner.graveyard.addCard(card)
        }
    }

    fun putIntoGraveyard(gameStatus: GameStatus, cardInstance: CardInstance) {
        putIntoGraveyard(gameStatus, listOf(cardInstance))
    }
}
